src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -202,21 +228,15 @@ Assembly TryLoadAssembly(AssemblyName assemblyName)
 
                     // Look in the MSBuild folder for any unresolved reference. It may be a dependency
                     // of MSBuild or a task.
-                    string targetAssembly = Path.Combine(msbuildPath, assemblyName.Name + ".dll");
-                    if (File.Exists(targetAssembly))
+                    foreach (string msbuildPath in msbuildSearchPaths)
                     {
-                        assembly = Assembly.LoadFrom(targetAssembly);
-                        loadedAssemblies.Add(assemblyName.FullName, assembly);
-                        return assembly;
-                    }
-
-                    // Finds and loads NuGet assemblies if msbuildPath is in a VS installation
-                    targetAssembly = Path.GetFullPath(Path.Combine(msbuildPath, "..", "..", "..", "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet", assemblyName.Name + ".dll"));
-                    if (File.Exists(targetAssembly))
-                    {
-                        assembly = Assembly.LoadFrom(targetAssembly);
-                        loadedAssemblies.Add(assemblyName.FullName, assembly);
-                        return assembly;
+                        string targetAssembly = Path.Combine(msbuildPath, assemblyName.Name + ".dll");
+                        if (File.Exists(targetAssembly))
+                        {
+                            assembly = Assembly.LoadFrom(targetAssembly);
```

Comment:
Not unique to this PR but should we improve error handling here? Like if LoadFrom fails because it exists but is a native DLL or something? Shouldn't be a major problem since the naming we're searching is very restricted.

Reply:
I haven't checked to see what (or how bad) the error message is for an invalid dll, but it would seem unnecessarily confusing to put out two assemblies with identical names and extensions in (at least roughly) the same spot.

Reply:
Yeah, let's just leave it.